### PR TITLE
Implement watchdog_exception rector

### PR DESCRIPTION
**Pull request by bbrala**
## Description
https://github.com/mglaman/drupal-change-record-triage/issues/557

## To Test
- Added tests ;) 

## Drupal.org issue
https://www.drupal.org/project/rector/issues/3400803

### CODE REVIEW

File: src/Rector/Deprecation/WatchdogExceptionRector.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    public function configure(array $configuration): void {
+        foreach ($configuration as $value) {
+            if (!($value instanceof DrupalIntroducedVersionConfiguration)) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Each configuration item must be an instance of "%s"',
+                    DrupalIntroducedVersionConfiguration::class
+                ));
+            }
+        }
```
Comment (mglaman) [inline src/Rector/Deprecation/WatchdogExceptionRector.php:36]: We should create an assert helper. Drupal core's `Inspector::assertAll`: https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Component/Assertion/Inspector.php#L53.

Maybe we could add a `require` on the following (or just copy relevant parts

```
"drupal/core-assertion": "^8 || ^9 || ^10 || ^11
```

Yeah.. let's just copy it into a new PR and simplify `configure`

File: src/Rector/Deprecation/WatchdogExceptionRector.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        parent::configure($configuration);
+    }
+
+    public function refactorWithConfiguration(Node $node, VersionedConfigurationInterface $configuration) {
+        if (!$node instanceof Node\Expr\FuncCall || $this->getName($node) !== 'watchdog_exception') {
+            return NULL;
+        }
+
+        assert($node instanceof Node\Expr\FuncCall);
```
Comment (mglaman) [inline src/Rector/Deprecation/WatchdogExceptionRector.php]: Redundant assert with above logic check

File: src/Rector/Deprecation/WatchdogExceptionRector.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+
```
Comment (mglaman) [inline src/Rector/Deprecation/WatchdogExceptionRector.php]: nit 2x lines

**Review by mglaman — APPROVED** [review]
Approving. Two nits and one idea

File: src/Rector/Deprecation/WatchdogExceptionRector.php (hunk 1 above)
Comment (bbrala) [reply src/Rector/Deprecation/WatchdogExceptionRector.php:36]: Good idea :)

**Pull request merged**